### PR TITLE
Refactor/tests

### PR DESCRIPTION
**Pull request by memsharded**
Changelog: Omit
Docs: Omit

The ``conans.test.utils.tools`` is getting a bit cluttered, too many things. Maybe refactoring and extracting some classes help reducing the low cohesion:

- Moving SCM related things from ``conans.test.utils.tools`` to ``conans.test.utils.scm``.
- Move some mocks/fake things from ``conans.test.utils.tools`` to ``conans.test.utils.mocks``

#tags: slow, svn

### CODE REVIEW

File: conans/test/utils/tools.py [hunk 1]
```diff
@@ -462,205 +452,8 @@ def package_revision_time(self, pref):
...
-        self._raise_if_non_interactive()
-        sub_dict = self.logins[remote_name]
-        index = self.login_index[remote_name]
-        tmp = sub_dict[index][1]
-        self.login_index.update([remote_name])
-        return tmp
-
-
 def _copy_cache_folder(target_folder):
+    _copy_cache_folder.master = dict()  # temp_folder(create_dir=False)
```
Comment (jgsogo) [inline conans/test/utils/tools.py]: Revert this change, this will clean and remove existing entries in the dictionary every time and will use a new folder each time.

**Review by jgsogo — CHANGES_REQUESTED** [review]
👍  I assume there are no source changes in the functions/classes moved from one file to the others.

File: conans/test/utils/tools.py (hunk 1 above)
Comment (memsharded) [reply conans/test/utils/tools.py]: Yes, fixed, this didn't belong to this PR, it was a half-baked thing.

Comment (memsharded) [reply conans/test/utils/tools.py]: Yes, absolutely no code changes, just moving things around.

**Review by jgsogo — APPROVED** [review]
